### PR TITLE
Rewrite all documentation

### PR DESCRIPTION
**Pull request by ecdye**
As has been mentioned many times, the openHABian documentation is poorly
written and often comes across in the wrong way. This is an attempt to
start fresh.

The goal is to keep the documentation clear, simple, and easy to follow.
All features should be fully documented and clear. Additionally the
expectations for what is supported should be clear, without making the
wrong impression on the end user, being overly wordy, or harsh.

Basically, make the documentation as great as the rest of openHABian.

### CODE REVIEW

**Comment by mstormi** [comment]
I believe you already recently 'softened' the HW requirements part and this https://github.com/openhab/openhabian/blob/main/docs/openhabian.md#3264-bit-support

What got lost along that way was that users should not be running 64 bit OS or java ob machines with 1GB.
Please prominently re-add a warning not to do that.

**Comment by ecdye** [comment]
@openhab/webui-maintainers Can you help me figure out how to make the sidebar have a dropdown to access a few subdocs like openHABian specific troubleshooting or more detailed feature guides.?

**Comment by florian-h05** [comment]
The sidebar structure is defined in the docs repo:
https://github.com/openhab/openhab-docs/blob/main/.vuepress/docs-sidebar.js

BTW, rather ping the doc maintainers instead of us UI maintainers.

Merry Christmas to you 🎄

**Comment by ecdye** [comment]
> The sidebar structure is defined in the docs repo:
> 
> https://github.com/openhab/openhab-docs/blob/main/.vuepress/docs-sidebar.js

Awesome, thanks!

> BTW, rather ping the doc maintainers instead of us UI maintainers.

Whoops, I'm sorry, I thought I even double checked, but I guess I wasn't looking close enough.

> Merry Christmas to you 🎄
Merry Christmas!

**Comment by stefan-hoehn** [comment]
> Can you help me figure out how to make the sidebar

Did you manage to create it or do you still need help? (I saw https://github.com/openhab/openhab-docs/pull/2434)

**Comment by ecdye** [comment]
> Did you manage to create it or do you still need help? (I saw [openhab/openhab-docs#2434](https://github.com/openhab/openhab-docs/pull/2434))

I think that I got it, I just need to finish making all the changes here first.

**Comment by ecdye** [comment]
OK @mstormi It is not perfect but it is all that I can convince myself to do for now. It's much better than it was.

**Comment by stefan-hoehn** [comment]
@ecdye I'll wait for @mstormi 's feedback, wdyt?

File: docs/openhabian-backup.md [hunk 1]
```diff
@@ -1,141 +1,148 @@
...
+In fact you should not ever do that!
+
+Reconfiguring openHAB and openHABian from scratch can be incredibly painful (trust me, I speak from experience).
+If you take a couple of basic precautions and implement some basic backup features when you setup openHABian, you can greatly reduce the chance that you would need to restart from scratch.
+
+We will provide some basic guidance below, please take the time to implement it, we promise you will regret it if you don't!
+
+## What to Prepare For
+
+In an ideal world, you should have a spare for *every* component that you use in you smart home system.
```
Comment (mstormi) [inline docs/openhabian-backup.md:19]:
```suggestion
In an ideal world, you should have a spare for *every* component that you use in your smart home system.
```

File: docs/openhabian-backup.md [hunk 2]
```diff
@@ -1,141 +1,148 @@
...
+This means having a backup SD card, RPi, router, switch, and any external addons (e.g. ZWave stick).
+
+Since you probably won't actually do that (but you should) here are some simple things that can reduce the headache if it is simply a part of your openHABian instance fails.
```
Comment (mstormi) [inline docs/openhabian-backup.md:22]: Double-insist on a controller spare (ZWave or Zigbee stick of same model)

File: docs/openhabian-backup.md [hunk 3]
```diff
@@ -1,141 +1,148 @@
...
+
+If you plan to mirror you SD card using the auto-backup features in openHAB, you should purchase 2 of the same model of SD card when you setup your system.
+A different model of SD card may work, but sometimes models differ slightly in size causing issues.
+If you can't find the same model, play it safe and buy a model larger than what you currently have to use for your backup.
+
+::: caution Moving the Root Filesystem
+Moving your system root to a USB stick or SSD is unsupported and dangerous.
+Besides USB sticks in particular suffer from the same issues as SD cards.
+SSD's do as well but to a lesser extent.
+In summary, just don't do it!
```
Comment (mstormi) [inline docs/openhabian-backup.md:41]: I know you don't want to write this but a major reason for the harshish old text is that there's still a number of oh users that boot from SSDs and keep insisting on telling other that to be a good idea.
While not wrong technically in itself, we know that they are neither implementing this generically nor are they supporting that so it's always us to pick up the slack.
We really must give more background info and make the reasoning clearer here, else too many inexperienced readers will again fall victim to those telling them SSDs are *the* solution.

File: docs/openhabian-backup.md [hunk 4]
```diff
@@ -1,141 +1,148 @@
...
+
+::: important
+You need to provide your storage *before* you install Amanda.
+:::
+
+You have to mount the USB stick or disk from your NAS to a directory on your openHABian box.
+If you don't know what this means in UNIX terms, please do some internet searches now to learn more.
+Your mountpoint should be a directory on your Raspberry Pi with the USB device or NAS mounted to.
+
+If you are a Windows user, please note that CIFS shares are not supported and almost certainly will not work.
```
Comment (mstormi) [inline docs/openhabian-backup.md:133]: Make this more prominent again.
Again, a major reason for the existing wording is that too many people have missed this info and tried CIFS. We don't want to end up back there. The quality of this documentation is also determined by how much it keeps off *ourselves*,
I don't want to repeatedly read and have to answer posts on the forum when people did it the wrong way because they were unwilling or to lazy to read the full docs (which is something that'll always be the case so we must male sure the relevant information is placed first inside a section ).

File: docs/openhabian-backup.md
[the remark this answers is not in the document]
Comment (ecdye) [reply docs/openhabian-backup.md:1]: Yep, I know. Thats one of the reasons I'm adding these to the sidebar to reduce any issues that occur as a result.

File: docs/openhabian-troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+If none of the things listed below work you are going to need to do more research on your own.
+Networking is highly complex and there are many different things that could be the issue.
+
+#### Wi-Fi
+
+WiFi requires user configuration prior to the first boot of openHABian.
+For more information on how to configure Wi-Fi before first boot see [Wi-Fi Settings](./openhabian.md#wi-fi-settings)
+
+If you would rather not try to do any additional configuration before first boot you can try to make use of the [hotspot](./openhabian.md#wi-fi-hotspot) feature.
```
[the remark this answers is not in the document]
Comment (ecdye) [reply docs/openhabian-troubleshooting.md:59]: Somewhat, but it is also partially because much of that content has been shifted around and moved to the main document and this references back to it.
Especially because much of it was a restatement of what was in the main install docs and has been clarified there.

I also removed a little bit of it because some of it was beyond the scope of what we really support anyways.

**Comment by ecdye** [comment]
@mstormi I've addressed most of your comments, please let me know if there are any other concerns.

File: docs/openhabian-troubleshooting.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (mstormi) [reply docs/openhabian-troubleshooting.md:63]: > but it is also partially because much of that content has been shifted around and moved to the main document

The DEBUG guide was meant to be a standalone document to point people with problems at, and specifically also those that ignored the official docs in the first place. I agree we should rewrite it but I also think we should stick to that idea.
It was written back when we did not have the hotspot and we did not have network manager.
Both have implications that we should explain a lot better, eventually including what you can do on the comitup console if it does not come up in the first place, or Network manager commands rather than `ifconfig`s
Plus mention static IPs prominently (that you must not attempt that and what to do to nonetheless obtain the same IP on every DHCP).

File: docs/openhabian-troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+#### 2+ GB of RAM
+
+You probably have enough RAM to run all the packages you want unless you are going crazy so you should probably keep reading to see what else could be going wrong.
+
+### Network Access
+
+openHABian requires direct Internet access for the duration of the installation.
+Ensure that your openHABian box has access to the internet and is not being blocked by any settings configured in your router.
+Both Ethernet and Wi-Fi are both supported at install time.
```
Comment (mstormi) [inline docs/openhabian-troubleshooting.md:53]:
```suggestion
Both Ethernet and Wi-Fi are supported at install time.
```

File: docs/openhabian-troubleshooting.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+If none of the things listed below work you are going to need to do more research on your own.
+Networking is highly complex and there are many different things that could be the issue.
+
+#### Wi-Fi
+
+WiFi requires user configuration prior to the first boot of openHABian.
```
Comment (mstormi) [inline docs/openhabian-troubleshooting.md:60]:
```suggestion
A non-interactive installation with static use of your home Wi-Fi requires user configuration action prior to the first boot of openHABian.
```

File: docs/openhabian-troubleshooting.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+#### 2+ GB of RAM
+
+You probably have enough RAM to run all the packages you want unless you are going crazy so you should probably keep reading to see what else could be going wrong.
+
+### Network Access
+
+openHABian requires direct Internet access for the duration of the installation.
+Ensure that your openHABian box has access to the internet and is not being blocked by any settings configured in your router.
+Both Ethernet and Wi-Fi are both supported at install time.
+
```
Comment (mstormi) [inline docs/openhabian-troubleshooting.md:54]:
```suggestion

openHABian will use Ethernet if connected.
If neither Ethernet nor [static Wi-Fi configuration](#Wi-Fi) work and the Internet cannot be reached, the installer will fire up a hotspot with a wireless LAN named `openhabian-<n>`.
Use your mobile to connect to it, to select your home Wi-Fi or to enter its name and openHABian will connect there permanently, i.e. safe across reboots.
```

**Review by mstormi — CHANGES_REQUESTED** [review]
If you merge my suggestions, I'm fine with merging the PR.

**Comment by ecdye** [comment]
@mstormi Sorry for the delay on this, I got back into school and haven't had the time to look at it again. I just merged all your suggestions. Let me know if there are any other issues, or feel free to merge.

**Comment by ecdye** [comment]
Also it appears that Build is broken on PRs right now, but I don't know why right now. We can go ahead and merge this and then I can investigate why it is not working, as it is clearly unrelated to the changes in this PR.

**Comment by mstormi** [comment]
yes okay

**Comment by mstormi** [comment]
I believe we have separate docs for troubleshooting and backup now but noone knows - we're missing pointers there from the main document.

Can you refactor this ?

**Comment by ecdye** [comment]
Yeah it's a bug with the docs page, I'm working with them to get it fixed in the sidebar
